cdap-tms/src/main/java/co/cask/cdap/messaging/store/hbase/HBaseTableFactory.java

```diff
@@ -136,6 +140,96 @@ public PayloadTable createPayloadTable(String tableName) throws IOException {
     );
   }
 
+  @Override
+  public void upgradeMessageTable(String tableName) throws IOException {
+    try {
+      upgradeCoProcessor(tableUtil.createHTableId(NamespaceId.SYSTEM, tableName),
+                         tableUtil.getMessageTableRegionObserverClassForVersion());
+    } catch (TableNotFoundException ex) {
```

Comment:
How about if the table was not disabled? It will thrown an `TableNotDisabledException`, right? Shall we handle it and check if the table needs to be disabled and perform the upgrade if needed?